src/Modules/UngradedModules/Methods.jl

```diff
@@ -233,16 +233,7 @@ ring_map(f::SubQuoHom) = f.ring_map
 #  re-evaluate and use or not
 
 function getindex(r::Hecke.SRow, u::AbstractUnitRange)
-  R = base_ring(r)
-  s = sparse_row(R)
-  shift = 1-first(u)
-  for (p,v) = r
-    if p in u
-      push!(s.pos, p+shift)
-      push!(s.values, v)
-    end
-  end
-  return s
+  return getindex(r, base_ring(r), u)
 end
 
 function getindex(r::Hecke.SRow, R::AbstractAlgebra.Ring, u::AbstractUnitRange)
```

Comment:
Note that this enables calls of the form `some_srow[R,1:r]` which I find really disturbing (this is how I learned about this function in the first place: I am touching a place calling it in PR #3696, inside `src/Modules/UngradedModules/Methods.jl`

Reply:
Sounds to me like someone (TM) should remove this `getindex` method? Why should one specify the ring for the result in a `getindex`?